### PR TITLE
chore: Migrate cluster-api from atmosphere operator to magnum role

### PR DESCRIPTION
**Pull request by okozachenko1203**

### CODE REVIEW

File: roles/openstack_helm_magnum/tasks/main.yml [hunk 1]
```diff
@@ -37,6 +37,19 @@
...
+    kubeconfig: /etc/kubernetes/admin.conf
+    state: present
+    template:
+      - path: 'templates/capi-bootstrap.yml.j2'
+      - path: 'templates/capi-control-plane.yml.j2'
+      - path: 'templates/capi-core.yml.j2'
+      - path: 'templates/capi-provider.yml.j2'
+      - path: 'templates/capi-rbac.yml.j2'
+  run_once: true
+  changed_when: false
```
Comment (okozachenko1203) [inline roles/openstack_helm_magnum/tasks/main.yml:51]: There are some fields changed whenever apply so need to set this false manually.

- CRDs include conversion.webhook.clientConfig.caBundle which is injected by cert-manager after deployment. So it is always different.
- Rules of aggregated clusterRoles are rendered after clusterRole is deployed. So it is always different.

File: roles/openstack_helm_magnum/tasks/main.yml [hunk 2]
```diff
@@ -358,15 +371,6 @@
               nodeSelector:
                 openstack-control-plane: enabled
 
-- name: Create Ingress
```
Comment (mnaser) [inline roles/openstack_helm_magnum/tasks/main.yml:361]: Is this accidentally removed?

File: roles/openstack_helm_magnum/tasks/main.yml (hunk 1 above)
Comment (mnaser) [reply roles/openstack_helm_magnum/tasks/main.yml:51]: I think there might be a Kubernetes setting to ignore fields on change?  Or was that Terraform?

**Review by mnaser — CHANGES_REQUESTED** [review]

File: roles/openstack_helm_magnum/tasks/main.yml (hunk 2 above)
Comment (okozachenko1203) [reply roles/openstack_helm_magnum/tasks/main.yml:361]: no, there were 2 duplicated tasks

File: roles/openstack_helm_magnum/tasks/main.yml (hunk 1 above)
Comment (okozachenko1203) [reply roles/openstack_helm_magnum/tasks/main.yml:51]: TF has but ansible doesn't

File: roles/openstack_helm_magnum/tasks/main.yml (hunk 2 above)
Comment (mnaser) [reply roles/openstack_helm_magnum/tasks/main.yml:361]: Are you sure?  One is for registry and one is for magnum..

Comment (okozachenko1203) [reply roles/openstack_helm_magnum/tasks/main.yml:361]: yes, i am sure. there were 2 ingresses for magnum before

**Review by mnaser — APPROVED** [review]

**Pull request merged**